### PR TITLE
fix #62

### LINKED ISSUE #62: Fails if $HOME/.mycroft doesn't exist
Currently the saving of skills.json doesn't validate that the folder `~/.mycroft` folder exists.

The [write_skill_state()](https://github.com/MycroftAI/mycroft-skills-manager/blob/master/msm/skill_state.py#L24) function should validate that the directory exists and try to create it if it doesn't.

### PR DESCRIPTION
**Pull request by luca-vercelli**
This is essentially https://github.com/MycroftAI/mycroft-skills-manager/pull/63/commits/6adf346c984bd1aa330239757ae158afafdfd125, taking in account of given suggestions

### CODE REVIEW

File: msm/skill_state.py [hunk 1]
```diff
@@ -23,6 +24,12 @@ def load_device_skill_state() -> dict:
 
 def write_device_skill_state(data: dict):
     """Write the device skill state to disk."""
+    dir_path = dirname(expanduser(SKILL_STATE_PATH))
+    try:
+        # create folder if it does not exist
+        makedirs(dir_path)
+    except:
```
Comment (forslund) [inline msm/skill_state.py]: bare exceptions are generally frowned upon, make this `except Exception:`

**Review by forslund — CHANGES_REQUESTED** [review]
Thanks, for contributing this fix. I flagged a minor caveat with the bare except. otherwise this looks good. To protect yourself and Mycroft we require you to sign a Contributor License Agreement (stating that you retain ownership of the contributed code but Mycroft is free to use it).

Please go to https://mycroft.ai/cla to start the process.

Once more thanks for picking up this issue!

**Comment by krisgesling** [comment]
CLA received and looks like the feedback has been addressed. Are you happy for this to get merged Ake?

**Comment by forslund** [comment]
Sorry for the late review. it works fine, thank you for taking care of this.

**Pull request merged**